tests/integration/security/authorization_test.go

```diff
@@ -1363,5 +1368,36 @@ func TestAuthorization_Custom(t *testing.T) {
 			}
 
 			rbacUtil.RunRBACTest(ctx, cases)
+
+			type ingressCase struct {
+				Name     string
+				Host     string
+				Path     string
+				IP       string
+				WantCode int
+			}
+			newIngressCase := func(Name string, Host string, Path string, IP string, WantCode int) ingressCase {
+				return ingressCase{
+					Name:     Name,
+					Host:     Host,
+					Path:     Path,
+					IP:       IP,
+					WantCode: WantCode,
+				}
+			}
+
+			ingressCases := []ingressCase{
+				newIngressCase("deny www.company.com", "www.company.com", "/", "172.16.0.1", 403),
```

Comment:
I don't think you should use the `newIngressCase`, just modify and use the same `newTestCase` to add the tests for ingress gateway.

Also you don't need to use the host or IP, just send the requests at different paths like other tests should be good enough.